addon/lib/shooter.js

```diff
@@ -551,16 +566,16 @@ const ShotContext = Class({
           this.shot.update(attrs);
         }, this)));
       watchPromise(allPromisesComplete(promises).then((function () {
-        return this.shot.save();
+        //return this.shot.save();
```

Comment:
Do:

```javascript
this._everythingSaved = watchPromise(allPromisesComplete(promises))
// later:
this._everythingSaved.then(() => {
  this.uploadData();
});
```

Then actual uploading will always block on everything being completed.  Note that allPromisesComplete resolves even when there is a failed promise.